javascript/ql/src/semmle/javascript/ApiGraphs.qll

```diff
@@ -912,11 +930,17 @@ module API {
     }
 
     /** Gets the API node for the `i`th parameter of this invocation. */
+    pragma[nomagic]
     Node getParameter(int i) {
       result = callee.getParameter(i) and
-      result.getARhs() = getArgument(i)
+      result = getAParameterCandidate(i)
     }
 
+    /**
+     * Gets an API node where a RHS of the node if the `i`th argument to this call.
+     */
+    private Node getAParameterCandidate(int i) { result.getARhs() = getArgument(i) }
```

Comment:
I'm a little bit surprised that pulling this predicate out makes a difference. I assume you have checked that the optimiser doesn't just inline it?

Reply:
> I assume you have checked that the optimiser doesn't just inline it? 

I had not. I only checked the clause timing reports.   
But I just checked, and the optimizer didn't inline it.

Reply:
:+1: